src/main/drivers/bus_spi.c

```diff
@@ -259,13 +261,22 @@ bool spiBusTransfer(const busDevice_t *bus, const uint8_t *txData, uint8_t *rxDa
 
 void spiSetDivisor(SPI_TypeDef *instance, uint16_t divisor)
 {
-#define BR_CLEAR_MASK 0xFFC7
+#define BR_BITS ((BIT(5) | BIT(4) | BIT(3)))
+
+    // SPI2 and SPI3 are always on APB1/AHB1 which PCLK is half that of APB2/AHB2.
+
+    if (instance == SPI2 || instance == SPI3) {
+        divisor /= 2; // Safe for divisor == 0 or 1
+    }
+
     SPI_Cmd(instance, DISABLE);
 
-    const uint16_t tempRegister = (instance->CR1 & BR_CLEAR_MASK);
+    const uint16_t tempRegister = (instance->CR1 & ~BR_BITS);
     instance->CR1 = (tempRegister | ((ffs(divisor | 0x100) - 2) << 3));
```

Comment:
There is `LL_SPI_SetBaudRatePrescaler` function (which only does masking and set)